corehq/apps/reports/standard/deployments.py

```diff
@@ -80,60 +79,62 @@ def headers(self):
     def selected_app_id(self):
         return self.request_params.get(SelectApplicationFilter.slug, None)
 
+    @memoized
```

Comment:
Do you have an appropriate cache timeout in mind. I am not particularly scared of an outdated cache, because I dont think app names change often but I do not have a sense of what is a reasonable amount of time to keep it.